lib/evmone/instructions.hpp

```diff
@@ -764,6 +764,24 @@ inline Result gas(StackTop stack, int64_t gas_left, ExecutionState& /*state*/) n
     return {EVMC_SUCCESS, gas_left};
 }
 
+inline void tload(StackTop stack, ExecutionState& state) noexcept
+{
+    const auto key = intx::be::store<evmc::bytes32>(stack.pop());
```

Comment:
`sload` gets a reference to `stack.top()` and assigns to it, without doing `push()`/`pop()`